util/memorydb/src/lib.rs

```diff
@@ -177,19 +186,12 @@ impl MemoryDB {
 	}
 }
 
-impl HashDB for MemoryDB {
-	fn get(&self, key: &H256) -> Option<DBValue> {
-		if key == &KECCAK_NULL_RLP {
-			return Some(DBValue::from_slice(&NULL_RLP));
-		}
-
-		match self.data.get(key) {
-			Some(&(ref d, rc)) if rc > 0 => Some(d.clone()),
-			_ => None
-		}
-	}
+impl<H: Hasher> HashDB<H> for MemoryDB<H> {
 
-	fn keys(&self) -> HashMap<H256, i32> {
+	// REVIEW: this method is what made it necessary to add a type param to H256FastMap, which I'd rather have avoided.
+	//         The problem is that the keys returned are `H256` and type inference fails on the `collect()` call.
+	//         I could not make it work without parameterizing H256FastMap too.
```

Comment:
note that `H256FastMap` is just a parameterization of `HashMap` with a custom hash-map hasher. We could allow the `Hasher` trait to specify a custom hash map hasher type and not have `MemoryDB` use `H256FastMap` directly.